scipy/stats/stats.py

```diff
@@ -6110,20 +6155,85 @@ def ttest_ind(a, b, axis=0, equal_var=True, nan_policy='propagate',
                                  alternative=alternative)
 
     else:
-        v1 = np.var(a, axis, ddof=1)
-        v2 = np.var(b, axis, ddof=1)
         n1 = a.shape[axis]
         n2 = b.shape[axis]
 
+        if trim == 0:
+            v1 = np.var(a, axis, ddof=1)
+            v2 = np.var(b, axis, ddof=1)
+            m1 = np.mean(a, axis)
+            m2 = np.mean(b, axis)
+        else:
+            #
+            v1, m1, n1 = _ttest_trim_var_mean_len(a, trim, n1, axis)
+            v2, m2, n2 = _ttest_trim_var_mean_len(b, trim, n2, axis)
+
         if equal_var:
             df, denom = _equal_var_ttest_denom(v1, n1, v2, n2)
         else:
             df, denom = _unequal_var_ttest_denom(v1, n1, v2, n2)
-        res = _ttest_ind_from_stats(np.mean(a, axis), np.mean(b, axis),
-                                    denom, df, alternative)
+
+        res = _ttest_ind_from_stats(m1, m2, denom, df, alternative)
     return Ttest_indResult(*res)
 
 
+def _ttest_trim_var_mean_len(a, trim, n, axis):
+    """Variance, mean, and length of winsorized input along specified axis"""
+    # for use with `ttest_ind` when trimming.
+    # further calculations in this test assume that the inputs are sorted.
+    # From [4] Section 1 "Let x_1, ..., x_n be n ordered observations..."
+    a = np.sort(a, axis=axis)
+
+    # `g_*` is the number of elements to be replaced on each tail, converted
+    # from a percentage amount of trimming
+    g = int(n * trim)
+
+    # Calculate the Winsorized variance of the input samples according to
+    # specified `g`
+    v = _calculate_winsorized_variance(a, g, n, axis)
+
+    # the total number of elements in the trimmed samples
+    n -= 2 * g
+
+    # calculate the g-times trimmed mean, as defined in [4] (1-1)
+    m = trim_mean(a, trim, axis=axis)
+    return v, m, n
+
+
+def _calculate_winsorized_variance(a, g, n, axis):
+    """Calculates g-times winsorized variance along specified axis"""
+    # it is expected that the input `a` is sorted along the correct axis
+    # move the intended axis to the end that way it is easier to manipulate
+    a = np.moveaxis(a, axis, -1)
+
+    # build a right and left tail to replace the trimmed values, fill new tails
+    # with the leftmost and rightmost value from the trimmed array. This can be
+    # see in effect in (1-3) in [4], where the leftmost and rightmost tails are
+    # replaced with `(g + 1) * x_{g + 1}` on the left and `(g + 1) * x_{n - g}`
+    # on the right. Zero-indexing turns `g + 1` to `g`, and `n - g` to
+    # `n - g - 1` in arraying splicing.
+    left_tail = np.repeat(a[..., [g]], g, axis=-1)
+    right_tail = np.repeat(a[..., [n - g - 1]], g, axis=-1)
+
+    # Winsorization and variance calculation are done in one step in [4] (1-3),
+    # but here winsorization is done first.
+    a_win = np.concatenate((left_tail, a[..., g: n - g], right_tail), axis=-1)
```

Comment:
If I understand correctly, consider copying `a` then replacing the values in the left and right tails. If you need it in here, get it from `a.shape[axis]`, but I think you don't need it - you can just use negative indices, e.g. this would be
```suggestion
    left_tail = np.repeat(a[..., [g]], g, axis=-1)
    right_tail = np.repeat(a[..., [n - g - 1]], g, axis=-1)

    # Winsorization and variance calculation are done in one step in [4] (1-3),
    # but here winsorization is done first.
    a_win = np.concatenate((left_tail, a[..., g: - g], right_tail), axis=-1)
```

Reply:
Actually, you don't have to copy `a` because that has been done already when you sorted. That makes me prefer the suggested approach even more. It becomes an in-place operation rather than requiring copying every element when you concatenate.